diff --git a/third_party/napi/include/napi.h b/third_party/napi/include/napi.h
index bc45b6c04..b8a0156b1 100644
--- a/third_party/napi/include/napi.h
+++ b/third_party/napi/include/napi.h
@@ -904,11 +904,11 @@ class NAPI_EXTERN ArrayBuffer : public Object {
 ///     }
 class NAPI_EXTERN TypedArray : public Object {
  public:
-  NAPI_INLINE TypedArray() : Object(), _type(unknown_array_type), _length(0) {}
+  NAPI_INLINE TypedArray() : Object(), _type(UnknownArrayType()), _length(0) {}
 
   NAPI_INLINE TypedArray(napi_env env, napi_value value)
       : Object(env, value),
-        _type(unknown_array_type),
+        _type(UnknownArrayType()),
         _length(0) {}  ///< Wraps a N-API value primitive.
 
   napi_typedarray_type TypedArrayType()
@@ -931,8 +931,9 @@ class NAPI_EXTERN TypedArray : public Object {
                          napi_typedarray_type type, size_t length)
       : Object(env, value), _type(type), _length(length) {}
 
-  static const napi_typedarray_type unknown_array_type =
-      static_cast<napi_typedarray_type>(-1);
+  static NAPI_INLINE napi_typedarray_type UnknownArrayType() {
+    return static_cast<napi_typedarray_type>(-1);
+  }
 };
 
 #define NAPI_FOR_EACH_TYPED_ARRAY(V)                      \
